### ISSUE TITLE
Calcite input clear button not focusable by keyboard navigation

### ISSUE THREAD

**Issue by JotaTe** [Esri/calcite-components#14520]
**monday.com sync:** #12131868790

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Which Component

Input

### Actual Behavior

When a Calcite input has the clearable property set to 'True', the clear button (when visible) is assigned a tabindex of -1 which makes it impossible to navigate to via the keyboard.

### Expected Behavior

The user should be able to navigate to the clear button (X) using keyboard navigation (tab).

### Reproduction Sample

https://codepen.io/minijcw/pen/ogYoymE

### Reproduction Steps

- Navigate to a Hub site. ([https://hub.arcgis.com/](https://hub.arcgis.com/ will do))
- Locate the "Search” input.
- Enter text into the search bar.
- Attempt to move keyboard focus to the "X" ("Clear value") button using the Tab key.
- Observe that the "X" button cannot be focused or activated with the keyboard.

### Reproduction Version

5.0.2

### Priority impact

impact - p3 - not time sensitive

### Impact

This has been identified as an a11y issue.

### Calcite package

- [ ] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/calcite-ui-icons
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

N/A

**Comment by JotaTe** [comment]
Adding a link back to our bug issue: [https://devtopia.esri.com/dc/hub/issues/15823](https://devtopia.esri.com/dc/hub/issues/15823)

**Comment by geospatialem** [comment]
This is an intentional design decision and in alignment with the [native HTML `input` when `type="search"`](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/input/search#try_it).

More information on the design in this comment: https://github.com/Esri/calcite-design-system/issues/9718#issuecomment-2243602182

> Since the component is clearable via the `esc` key when `clearable` is set on the component, the button does not need to be keyboard accessible. Closing the above out as out of scope.

**Comment by tom12993** [comment]
@geospatialem  The accessibility team flagged this as an a11y violation on ArcGIS Monitor. Is there any validity to this, or should we dismiss it since it's expected behavior?

```
# Violation:
Interactive elements need a descriptive label and keyboard focus

# Severity:
10

# Media Type:
Screen reader/Keyboard Accessibility

# Issue
The "Clear title" control is not keyboard accessible and does not have a descriptive name.

# User Impact
The user does not know this control exists.

# Recommendation
Change aria-label to "Clear title".
Add audio confirmation after button is pressed that title has been cleared.
Add button to tab order.
```

**Comment by geospatialem** [comment]
> [@geospatialem](https://github.com/geospatialem) The accessibility team flagged this as an a11y violation on ArcGIS Monitor. Is there any validity to this, or should we dismiss it since it's expected behavior?

Correct, this is invalid and expected behavior - please dismiss. I've pinged the accessibility team to notify the testers this is expected, and native web behavior for future testing across products. Appreciate you bringing this to our attention.

Closing out the above as unplanned.

**Comment by tom12993** [comment]
@geospatialem Thank you!
